src/aedat4.hpp

```diff
@@ -266,6 +271,106 @@ struct AEDAT4 {
     }
   }
 
+  static std::tuple<char *, size_t> compress_lz4(char *buffer, size_t size) {
+    auto lz4FrameBound = LZ4F_compressBound(size, nullptr);
+    auto new_buffer = new char[lz4FrameBound];
+
+    auto compression =
+        LZ4F_compressFrame(new_buffer, lz4FrameBound, buffer, size, nullptr);
+    return {new_buffer, compression};
+    // LZ4F_compressionContext_t ctx;
+    // auto contextCreation = LZ4F_createCompressionContext(&ctx, LZ4F_VERSION);
+    // auto headerSize =
+    //     LZ4F_compressBegin(ctx, new_buffer, lz4FrameBound, nullptr);
+    // if (LZ4F_isError(headerSize)) {
+    //   std::stringstream err;
+    //   err << "Compression error: " << LZ4F_getErrorName(headerSize);
+    //   throw std::runtime_error(err.str());
+    // }
+
+    // auto packetSize = LZ4F_compressUpdate(ctx, new_buffer, lz4FrameBound,
+    //                                       buffer, size, nullptr);
+    // auto footerSize = LZ4F_compressEnd(ctx, new_buffer, lz4FrameBound,
+    // nullptr);
+    // return {new_buffer, headerSize + packetSize + footerSize};
```

Comment:
Why the commented code and how are errors handled?

Reply:
The block is commented because we're not using the XML at the moment. It's also not something I'm including in the generated `.aedat4` files at the moment, so including the block breaks the reading of AEDAT files without the XML nodes.